AK/Optional.h

```diff
@@ -136,7 +136,7 @@ class [[nodiscard]] OptionalBase {
 };
```

Comment:
Should we always use `Traits<T>::optional_has_value`? If it covers more cases, having a single way of doing it would simplify the implementation a bit.

Reply:
Can you elaborate on this?

Reply:
> Can you elaborate on this?
I misunderstood how it works, forget it.